src/Browser/GoogleChrome/VersionResolver.php

```diff
@@ -58,8 +61,18 @@ private function getVersionFromCommandLine(string $command) : Version
         return Version::fromString($process->getOutput());
     }
 
+    private function getProcess(string $command) : Process
+    {
+        return $this->processes[$command] ?? Process::fromShellCommandline($command);
+    }
+
     public function supports(BrowserName $browserName) : bool
     {
         return $browserName->equals(BrowserName::GOOGLE_CHROME());
     }
+
+    public function setProcess(string $command, Process $process) : void
```

Comment:
I understand why you've done this, but I am against adding these setter and getter just to make it testable. Seeing this made me realize there is something wrong on a deeper level.

The Process class makes use of an implicit dependency: the command line environment.

Just like the Filesystem is a dependency, the command line environment is also a dependency. This implicit dependency is what's making this code hard to test.

In order to solve this, I propose adding a CommandLineEnvironment interface with implementations like UnixCommandLineEnvironment and WindowsCommandLineEnvironment. The VersionResolver can depend on this new interface, which is now mockable and thus testable.

If you are up to it, you can give it a shot. If not, no problem. You can revert the changes in this file and remove the unit test and I will update this later.

Reply:
Yes, you are right.
I even considered after writing this pull request to have instead a ProcessFactory.

For the moment I added interface `CommandLineEnvironment` and `ShellCommandLineEnvironment`
i did not split yet between Unix and Windows because we can use the Shell in both case but it might be necessary in the future